regex/src/main/scala/regex/Regex.scala

```diff
@@ -8,20 +8,65 @@ package regex
 
 // Add your definitions here
 
+trait RegularLanguage
+
+case object Empty extends RegularLanguage
+
+case object Epsilon extends RegularLanguage
+
+case class Character(char: Char) extends RegularLanguage
+
+case class Union(l1: RegularLanguage, l2: RegularLanguage) extends RegularLanguage
+
+case class Concat(l1: RegularLanguage, l2: RegularLanguage) extends RegularLanguage
+
+case class Star(l: RegularLanguage) extends RegularLanguage
+
 /** *****************************************************************************
   * Derivatives
   *
   * Fill in the function definitions below
   */
 
 /** Simplifies a regular language */
-def simplify(lang: RegularLanguage): RegularLanguage = ???
+def simplify(lang: RegularLanguage): RegularLanguage = lang match
+  case Concat(Epsilon, lang) => simplify(lang)
```

Comment:
Using lang here results in the code being more intuitive, but it could also be confused with the argument lang in line 32. I still think it's a good idea, though